File: android/app/src/main/java/com/sproutly/app/core/design/Components.kt
package com.sproutly.app.core.design

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

@Composable
fun SproutlyCard(
    modifier: Modifier = Modifier,
    accent: Boolean = false,
    content: @Composable ColumnScope.() -> Unit,
) {
    val shape = RoundedCornerShape(20.dp)
    Column(
        modifier = modifier
            .clip(shape)
            .background(if (accent) BgElevated else BgSurface)
            .border(1.dp, Divider, shape)
            .padding(18.dp),
        content = content,
    )
}

@Composable
fun SectionLabel(text: String) {
    Text(
        text = text.uppercase(),
        color = TextMuted,
        style = MaterialTheme.typography.labelMedium,
    )
}

@Composable
fun MintPillButton(
    label: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    Button(
        onClick = onClick,
        enabled = enabled,
        modifier = modifier.heightIn(min = 52.dp),
        shape = RoundedCornerShape(16.dp),
        colors = ButtonDefaults.buttonColors(
            containerColor = LeafMint,
            contentColor = BgDeep,
        )
    ) {
        Text(label, fontWeight = FontWeight.SemiBold)
    }
}

@Composable
fun GhostButton(
    label: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    OutlinedButton(
        onClick = onClick,
        modifier = modifier.heightIn(min = 52.dp),
        shape = RoundedCornerShape(16.dp),
        colors = ButtonDefaults.outlinedButtonColors(contentColor = TextPrimary)
    ) {
        Text(label)
    }
}

@Composable
fun Chip(label: String, selected: Boolean, onClick: () -> Unit) {
    val bg = if (selected) LeafMint else Color.Transparent
    val fg = if (selected) BgDeep else TextPrimary
    val border = if (selected) LeafMint else Divider
    val shape = RoundedCornerShape(50)
    Box(
        modifier = Modifier
            .clip(shape)
            .background(bg)
            .border(1.dp, border, shape)
            .clickable(onClick = onClick)
            .padding(horizontal = 14.dp, vertical = 8.dp),
        contentAlignment = Alignment.Center,
    ) {
        Text(label, color = fg, style = MaterialTheme.typography.labelLarge)
    }
}

File: ios/Sproutly/Core/Design/Components.swift
import SwiftUI

struct SproutlyCard<Content: View>: View {
    var accent: Bool = false
    @ViewBuilder var content: () -> Content

    var body: some View {
        VStack(alignment: .leading, spacing: 12, content: content)
            .padding(18)
            .background(accent ? Palette.bgElevated : Palette.bgSurface)
            .overlay(
                RoundedRectangle(cornerRadius: 20).stroke(Palette.divider, lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: 20))
    }
}

struct SectionLabel: View {
    let text: String
    var body: some View {
        Text(text.uppercased())
            .font(SproutlyType.labelMedium)
            .kerning(0.5)
            .foregroundColor(Palette.textMuted)
    }
}

struct MintPillButton: View {
    let label: String
    var enabled: Bool = true
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(label)
                .font(SproutlyType.titleMedium.weight(.semibold))
                .frame(maxWidth: .infinity, minHeight: 52)
        }
        .disabled(!enabled)
        .background(enabled ? Palette.leafMint : Palette.leafMint.opacity(0.5))
        .foregroundColor(Palette.bgDeep)
        .clipShape(RoundedRectangle(cornerRadius: 16))
    }
}

struct GhostButton: View {
    let label: String
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(label)
                .font(SproutlyType.titleMedium)
                .frame(maxWidth: .infinity, minHeight: 52)
        }
        .foregroundColor(Palette.textPrimary)
        .overlay(
            RoundedRectangle(cornerRadius: 16).stroke(Palette.divider, lineWidth: 1)
        )
    }
}

struct SproutlyChip: View {
    let label: String
    let selected: Bool
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            Text(label)
                .font(SproutlyType.labelLarge)
                .padding(.horizontal, 14)
                .padding(.vertical, 8)
                .foregroundColor(selected ? Palette.bgDeep : Palette.textPrimary)
                .background(selected ? Palette.leafMint : Color.clear)
                .overlay(
                    Capsule().stroke(selected ? Palette.leafMint : Palette.divider, lineWidth: 1)
                )
                .clipShape(Capsule())
        }
        .buttonStyle(.plain)
    }
}

struct ScreenBackground<Content: View>: View {
    @ViewBuilder var content: () -> Content
    var body: some View {
        ZStack {
            Palette.bgDeep.ignoresSafeArea()
            content()
        }
    }
}
